internal/service/appfabric/ingestion.go

```diff
@@ -0,0 +1,315 @@
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"arn": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"app_bundle_identifier": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"app_bundle_arn": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			names.AttrID: framework.IDAttribute(),
+			"ingestion_type": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"state": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+			"tenant_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+func (r *resourceIngestion) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().AppFabricClient(ctx)
+
+	var plan resourceIngestionData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := &appfabric.CreateIngestionInput{
+		App:                 aws.String(plan.App.ValueString()),
+		AppBundleIdentifier: aws.String(plan.AppBundleIdentifier.ValueString()),
+		IngestionType:       awstypes.IngestionType(plan.IngestionType.ValueString()),
+		TenantId:            aws.String(plan.TenantId.ValueString()),
+		Tags:                getTagsIn(ctx),
+	}
+
+	out, err := conn.CreateIngestion(ctx, in)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.AppFabric, create.ErrActionCreating, ResNameIngestion, plan.App.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	if out == nil || out.Ingestion == nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.AppFabric, create.ErrActionCreating, ResNameIngestion, plan.App.String(), nil),
+			errors.New("empty output").Error(),
+		)
+		return
+	}
+
+	plan.App = flex.StringToFramework(ctx, out.Ingestion.App)
+	plan.AppBundleArn = flex.StringToFramework(ctx, out.Ingestion.AppBundleArn)
+	plan.AppBundleIdentifier = flex.StringToFramework(ctx, out.Ingestion.AppBundleArn)
+	plan.ARN = flex.StringToFramework(ctx, out.Ingestion.Arn)
+	plan.ID = types.StringValue(createIngestionID(string(*out.Ingestion.AppBundleArn), string(*out.Ingestion.Arn)))
+	plan.IngestionType = flex.StringToFramework(ctx, aws.String(string(out.Ingestion.IngestionType)))
+	plan.State = flex.StringToFramework(ctx, aws.String(string(out.Ingestion.State)))
+	plan.TenantId = flex.StringToFramework(ctx, out.Ingestion.TenantId)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *resourceIngestion) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
We need to revisit this a bit, consider looking at other update services like bedrock / bedrockagent examples and refactor if possible plz.

Reply:
Good callout...the AppFabric resources don't have specific update calls so, I was following the advice from Skaff about using the same logic in create.  Bedrock Agent has a specific update call.  What is the recommendation for services without an update, or do you have a similar service I could look at?